01-getting-started/projects/final/app/src/main/java/com/raywenderlich/podplay/ui/PodcastDetailsFragment.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2022 Razeware LLC
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *   Notwithstanding the foregoing, you may not use, copy, modify, merge, publish,
+ *   distribute, sublicense, create a derivative work, and/or sell copies of the
+ *   Software in any work that is designed, intended, or marketed for pedagogical or
+ *   instructional purposes related to programming, coding, application development,
+ *   or information technology.  Permission for such use, copying, modification,
+ *   merger, publication, distribution, sublicensing, creation of derivative works,
+ *   or sale is expressly withheld.
+ *
+ *   This project and source code may use libraries or frameworks that are
+ *   released under various Open-Source licenses. Use of those libraries and
+ *   frameworks are governed by their own individual licenses.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ */
+
+package com.raywenderlich.podplay.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.text.method.ScrollingMovementMethod
+import android.view.*
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.raywenderlich.podplay.R
+import com.raywenderlich.podplay.adapter.EpisodeListAdapter
+import com.raywenderlich.podplay.databinding.FragmentPodcastDetailsBinding
+import com.raywenderlich.podplay.util.GlideApp
+import com.raywenderlich.podplay.viewmodel.PodcastViewModel
+
+class PodcastDetailsFragment : Fragment(), EpisodeListAdapter.EpisodeListAdapterListener {
+
+  private val podcastViewModel: PodcastViewModel by activityViewModels()
+  private lateinit var databinding: FragmentPodcastDetailsBinding
```

Comment:
Probably I missed previous implementation of data binding somewhere, so make sure you update the whole project.

Data binding MUST always be cleaned up. So, this should be optional and once the fragment is destroyed, it should be set as null. The value is assigned properly. :]

Reply:
Agreed. Should update this.

Reply:
Assigned @vguzzi - https://trello.com/c/MvAOAIy5/26-ensure-databinding-is-destructed